internal/pipeline/executor.go

```diff
@@ -283,18 +288,22 @@ func filterMetaData(resources []*yaml.RNode) []*yaml.RNode {
 
 // fnFilters returns chain of functions that are applicable
 // to a given pipeline.
-func fnFilters(_ *Pipeline) []kio.Filter {
-	fn := &annotator{
-		key:   "builtin/setter-1",
-		value: "test",
+func fnFilters(p *Pipeline) ([]kio.Filter, error) {
+	filters, err := p.fnChain()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get function chain: %w", err)
 	}
 
-	// TODO(droot): Implement this with the logic to create
-	// function chain from a pipeline
-	// hardcoding built-in set-annotation function for testing
-	return []kio.Filter{
-		&fnRunner{
-			fn: fn,
-		},
+	var ans []kio.Filter
+	for _, f := range filters {
```

Comment:
It's not clear to me why we need this extra loop given that fnRunner implements `kio.Filter`.

Reply:
I am not familiar wit this part in Go. `filters` is in type `[]*fnRunner` and we need to return type `[]kio.Filter`. Type assertion doesn't work with this error message: `invalid type assertion: filters.([]kio.Filter) (non-interface type []*fnRunner on left)`. If I use `fns = append(fns, filters...)` the error message will be `cannot use filters (type []*fnRunner) as type []kio.Filter in append`.

I don't know why a loop can work :(

Reply:
I think I know now why this won't work unless in all the functions we use kio.Filter instead of fnRunner.

Not a big deal. Lets get on a call to try that.